src/profile/components/basicData/__tests__/BasicData.test.tsx

```diff
@@ -181,4 +184,75 @@ describe('<BasicData /> ', () => {
       await verifyValues(getTextOrInputValue, initialProfile);
     });
   });
+  it('invalid values are indicated and setting a valid value removes error', async () => {
+    await act(async () => {
+      const { clickElement, setInputValue, getElement } = await initTests();
+      const formFields = getFormFields(basicDataType);
+      await clickElement(editButtonSelector);
+      const testRuns = [
+        {
+          validData: basicData,
+          invalidData: { ...basicData, firstName: '' },
+          elementSelector: { id: 'basic-data-firstName' },
+          errorSelector: {
+            id: 'basic-data-firstName-helper',
+          },
+        },
+        {
+          validData: basicData,
+          invalidData: { ...basicData, lastName: '' },
+          elementSelector: { id: 'basic-data-lastName' },
+          errorSelector: {
+            id: 'basic-data-lastName-helper',
+          },
+        },
+        {
+          validData: basicData,
+          invalidData: {
+            ...basicData,
+            nickname: String('a').repeat(
+              (formFields.nickname.max as number) + 1
+            ),
+          },
+          elementSelector: { id: 'basic-data-nickname' },
+          errorSelector: {
+            id: 'basic-data-nickname-helper',
+          },
+        },
+      ];
+
+      // cannot use forEach with async/await
+      for (const runProps of testRuns) {
+        const {
+          validData,
+          invalidData,
+          elementSelector,
+          errorSelector,
+        } = runProps;
+        const elementGetter = () => getElement(elementSelector);
+        const errorElementGetter = () => getElement(errorSelector);
+
+        // set invalid values
+        await setValues(setInputValue, invalidData);
+        // submit also validates the form
+        await clickElement(submitButtonSelector);
+        await waitForElementAttributeValue(
+          elementGetter,
+          'aria-invalid',
+          'true'
+        );
+        // getElement throws is element is not found
+        // so element is not checked in expect
+        errorElementGetter();
```

Comment:
This probably works already as it is, but could we use this here in addition:

`expect(() => ...error...).not.toThrow(error)`?

Reply:
This is probably clearer

Reply:
Done.